File: sessions_mount_dir/cdzcpk0m78430xqx44gjqsbox/frontend/src/api/members.js
import axios from 'axios';

const API_BASE_URL = import.meta.env.VITE_API_BASE_URL || 'http://localhost:8000';

/**
 * 获取会员列表
 * @param {Object} params - 查询参数
 * @returns {Promise} Axios promise
 */
export const getMembers = (params = {}) => {
  return axios.get(`${API_BASE_URL}/api/v1/members`, { params });
};

/**
 * 创建新会员
 * @param {Object} memberData - 会员数据
 * @returns {Promise} Axios promise
 */
export const createMember = (memberData) => {
  return axios.post(`${API_BASE_URL}/api/v1/members`, memberData);
};

/**
 * 获取单个会员详情
 * @param {number} memberId - 会员ID
 * @returns {Promise} Axios promise
 */
export const getMemberById = (memberId) => {
  return axios.get(`${API_BASE_URL}/api/v1/members/${memberId}`);
};

/**
 * 更新会员信息
 * @param {number} memberId - 会员ID
 * @param {Object} memberData - 更新的会员数据
 * @returns {Promise} Axios promise
 */
export const updateMember = (memberId, memberData) => {
  return axios.put(`${API_BASE_URL}/api/v1/members/${memberId}`, memberData);
};

/**
 * 删除会员
 * @param {number} memberId - 会员ID
 * @returns {Promise} Axios promise
 */
export const deleteMember = (memberId) => {
  return axios.delete(`${API_BASE_URL}/api/v1/members/${memberId}`);
};

/**
 * 获取会员支付记录
 * @param {number} memberId - 会员ID
 * @returns {Promise} Axios promise
 */
export const getMemberPayments = (memberId) => {
  return axios.get(`${API_BASE_URL}/api/v1/members/${memberId}/payments`);
};

/**
 * 获取会员预约记录
 * @param {number} memberId - 会员ID
 * @returns {Promise} Axios promise
 */
export const getMemberBookings = (memberId) => {
  return axios.get(`${API_BASE_URL}/api/v1/members/${memberId}/bookings`);
};

/**
 * 更新会员卡状态
 * @param {number} memberId - 会员ID
 * @param {string} status - 新的卡状态
 * @returns {Promise} Axios promise
 */
export const updateMemberCardStatus = (memberId, status) => {
  return axios.patch(`${API_BASE_URL}/api/v1/members/${memberId}/card-status`, { status });
};
  return axios.get(`${API_BASE_URL}/api/v1/members/${memberId}/payments`);
};

File: sessions_mount_dir/3mftwi8l5k2ka8yxu3knqzik5/backend/app/api/v1/endpoints/members.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import List, Optional

from app.api.v1.api import get_db
from app.crud import crud_member
from app.schemas.member import Member, MemberCreate, MemberUpdate, MemberResponse
from app.models.member import MemberStatus

router = APIRouter()


@router.post("/", response_model=MemberResponse)
def create_member(
    *,
    db: Session = Depends(get_db),
    member_in: MemberCreate
):
    """
    Create a new member.
    """
    # Check if card number already exists
    member = crud_member.get_member_by_card_number(db, card_number=member_in.card_number)
    if member:
        raise HTTPException(
            status_code=400,
            detail="The member with this card number already exists in the system.",
        )
    member = crud_member.create_member(db=db, member=member_in)
    return member


@router.get("/", response_model=List[MemberResponse])
def read_members(
    db: Session = Depends(get_db),
    skip: int = Query(0, ge=0),
    limit: int = Query(100, ge=0),
    status: Optional[MemberStatus] = None
):
    """
    Retrieve members.
    """
    if status:
        members = crud_member.get_members_by_status(db, status=status, skip=skip, limit=limit)
    else:
        members = crud_member.get_members(db, skip=skip, limit=limit)
    return members


@router.get("/{member_id}", response_model=MemberResponse)
def read_member(
    *,
    db: Session = Depends(get_db),
    member_id: int
):
    """
    Get a specific member by ID.
    """
    member = crud_member.get_member(db, member_id=member_id)
    if not member:
        raise HTTPException(status_code=404, detail="Member not found")
    return member


@router.put("/{member_id}", response_model=MemberResponse)
def update_member(
    *,
    db: Session = Depends(get_db),
    member_id: int,
    member_in: MemberUpdate
):
    """
    Update a member.
    """
    member = crud_member.update_member(db, member_id=member_id, member=member_in)
    if not member:
        raise HTTPException(status_code=404, detail="Member not found")
    return member


@router.delete("/{member_id}", response_model=MemberResponse)
def delete_member(
    *,
    db: Session = Depends(get_db),
    member_id: int
):
    """
    Delete a member.
    """
    member = crud_member.delete_member(db, member_id=member_id)
    if not member:
        raise HTTPException(status_code=404, detail="Member not found")
    return member
